presto-test-jdbc-compatibility/src/test/java/io/prestosql/TestJdbcCompatibility.java

```diff
@@ -0,0 +1,528 @@
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.Array;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static com.google.common.base.Strings.repeat;
+import static com.google.common.base.Verify.verify;
+import static java.lang.Integer.min;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestJdbcCompatibility
+{
+    private static final Optional<Integer> DRIVER_VERSION_TESTED = Optional.ofNullable(System.getenv("PRESTO_JDBC_VERSION")).map(Integer::valueOf);
+
+    private static final int DEFAULT_PRECISION = 3; // milliseconds
+    private static final int MAX_SQL_PRECISION = 9; // nanoseconds
+    private static final int MAX_TIMESTAMP_PRECISION = 12; // picoseconds
+    private static final String TESTED_TZ = "Australia/Eucla"; // GMT+8:45
+
+    private TestingPrestoServer server;
+    private String serverUrl;
+
+    @BeforeClass
+    public void setup()
+    {
+        Logging.initialize();
+
+        server = TestingPrestoServer.builder()
+                .build();
+
+        serverUrl = format("jdbc:presto://%s", server.getAddress());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        server.close();
+    }
+
+    @Test
+    public void ensureProperDriverVersionLoaded()
+    {
+        if (DRIVER_VERSION_TESTED.isEmpty()) {
+            throw new SkipException("PRESTO_JDBC_VERSION environment variable is missing");
+        }
+
+        assertThat(PrestoDriver.class.getPackage().getImplementationVersion())
+                .isEqualTo(Integer.toString(DRIVER_VERSION_TESTED.get()));
+    }
+
+    @Test
+    public void ensureUsingProperCliVersion()
+            throws ClassNotFoundException
+    {
+        if (DRIVER_VERSION_TESTED.isEmpty()) {
+            throw new SkipException("PRESTO_JDBC_VERSION environment variable is missing");
+        }
+
+        assertThat(Class.forName("io.prestosql.jdbc.$internal.client.StatementClientFactory").getPackage().getImplementationVersion())
+                .isEqualTo(Integer.toString(DRIVER_VERSION_TESTED.get()));
+    }
+
+    @Test
+    public void testLongPreparedStatement()
+            throws Exception
+    {
+        String sql = format("SELECT '%s' = '%s'", repeat("x", 100_000), repeat("y", 100_000));
+
+        try (ResultSet rs = runQuery(sql)) {
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getBoolean(1)).isFalse();
+            assertThat(rs.next()).isFalse();
+        }
+    }
+
+    @Test
+    public void testTimestampSelect()
```

Comment:
testSelectTimestamp

Reply:
bump; 

you have this for `testSelectParametricTimestamp`, but not here
also, `testTimestampWithTimeZoneSelect` needs update as well